### PR TITLE
Feature mes 5012 fault count and summary provider home tests

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES 5012

Added fault count and fault summary providers and a suite of unit tests for both, for the home test categories (F, G, H, K) 

There will be more category specific logic added in later PRs.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by edwardwilson — APPROVED** [review]

File: src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    code: 'string',
+    description: 'string',
+    outcome: 'DF',
+  }] as QuestionResult[],
+  showMeQuestions: [{
+    code: 'string',
+    description: 'string',
+    outcome: 'DF',
+  }] as QuestionResult[],
+};
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts]: Not necessary as part of this PR but we should consider extracting these out into separate files which can be used be multiple categories. They are the same across the categories right? Would have to add a union type if we want to keep the typing but shouldn't be a problem.

File: src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      outcome: 'DF',
+    }] as QuestionResult[],
+    showMeQuestions: [{
+      code: 'string',
+      description: 'string',
+      outcome: 'DF',
+    }] as QuestionResult[],
+  },
+};
+export const vehicleChecksNoFaults:CatFUniqueTypes.VehicleChecks = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts]: Should be a space after the colon and before the type.

File: src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    outcome: 'P',
+  }] as QuestionResult[],
+  showMeQuestions: [{
+    code: 'string',
+    description: 'string',
+    outcome: 'P',
+  }] as QuestionResult[],
+};
+
+export const vehicleChecksTwoFaults:CatFUniqueTypes.VehicleChecks = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts]: Should be a space after the colon and before the type.

File: src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    }] as QuestionResult[],
+    showMeQuestions: [{
+      code: 'string',
+      description: 'string',
+      outcome: 'DF',
+    }] as QuestionResult[],
+  },
+};
+
+export const vehicleChecksNoFaults = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts]: Type?

File: src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    outcome: 'P',
+  }] as QuestionResult[],
+  showMeQuestions: [{
+    code: 'string',
+    description: 'string',
+    outcome: 'P',
+  }] as QuestionResult[],
+};
+
+export const vehicleChecksTwoFaults = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts]: Type?

File: src/providers/fault-count/__mocks__/cat-H-test-data-state-object.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    }] as QuestionResult[],
+    showMeQuestions: [{
+      code: 'string',
+      description: 'string',
+      outcome: 'DF',
+    }] as QuestionResult[],
+  },
+};
+
+export const vehicleChecksNoFaults = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-H-test-data-state-object.ts]: Type?

File: src/providers/fault-count/__mocks__/cat-H-test-data-state-object.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    outcome: 'P',
+  }] as QuestionResult[],
+  showMeQuestions: [{
+    code: 'string',
+    description: 'string',
+    outcome: 'P',
+  }] as QuestionResult[],
+};
+
+export const vehicleChecksTwoFaults = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-H-test-data-state-object.ts]: Type?

File: src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    }] as QuestionResult[],
+    showMeQuestions: [{
+      code: 'string',
+      description: 'string',
+      outcome: 'DF',
+    }] as QuestionResult[],
+  },
+};
+
+export const vehicleChecksNoFaults = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts]: Type?

File: src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    outcome: 'P',
+  }] as QuestionResult[],
+  showMeQuestions: [{
+    code: 'string',
+    description: 'string',
+    outcome: 'P',
+  }] as QuestionResult[],
+};
+
+export const vehicleChecksTwoFaults = {
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts]: Type?

File: src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    uphillStartDesignatedStart: true,
+  },
+  ETA: {
+    physical: false,
+    verbal: false,
+  },
+  eco: {
+    adviceGivenControl: false,
+    adviceGivenPlanning: false,
+  },
```
Comment (mgoldson) [inline src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts:25]: Not sure if it makes any difference but is controlledStop missing?

File: src/providers/fault-summary/cat-home-test/__tests__/fault-summary.cat-home-test.spec.ts [hunk 11]
```diff
@@ -0,0 +1,319 @@
...
+
+describe('FaultSummaryCatHomeTestHelper', () => {
+
+  configureTestSuite(() => {
+    TestBed.configureTestingModule({
+    });
+  });
+
+  describe('getDrivingFaultsCatHomeTest (No VehicleChecks)', () => {
+    it(`should return 3 driving fault count for category F`, () => {
```
Comment (mgoldson) [inline src/providers/fault-summary/cat-home-test/__tests__/fault-summary.cat-home-test.spec.ts]: No need to use back ticks on this test description and those below

**Review by mgoldson — APPROVED** [review]
A few minor optional things that you may choose to address but Looks good.
I guess as you've added more comprehensive tests in the helpers, there is no need to update the fault-count.spec.ts or fault-summary.spec.ts?

File: src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts (hunk 1 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts]: I think there may be a major refactoring into an npm package for this in future, so i'll hold off on doing this for now.

File: src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts (hunk 2 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts]: added

File: src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts (hunk 3 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-F-test-data-state-object.ts]: added

File: src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts]: added

File: src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts (hunk 5 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-G-test-data-state-object.ts]: added

File: src/providers/fault-count/__mocks__/cat-H-test-data-state-object.ts (hunk 7 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-H-test-data-state-object.ts]: added

File: src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts (hunk 8 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts]: added

File: src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts (hunk 9 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts]: added

File: src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts (hunk 10 above)
Comment (rossfauldsbjss) [reply src/providers/fault-count/__mocks__/cat-K-test-data-state-object.ts:25]: Great spot. Added controlledStop just as an empty object so it doesn't affect tests.

File: src/providers/fault-summary/cat-home-test/__tests__/fault-summary.cat-home-test.spec.ts (hunk 11 above)
Comment (rossfauldsbjss) [reply src/providers/fault-summary/cat-home-test/__tests__/fault-summary.cat-home-test.spec.ts]: Copy and paste when I was using test cases that required value substitution, removed the substitution but didn't remove back ticks. Great spot.

**Pull request merged**